build-system/tasks/changelog.js

```diff
@@ -117,25 +128,36 @@ function buildChangelog(gitMetadata, githubMetadata) {
   return gitMetadata;
 }
 
+/**
+ * Get the latest git tag from either a normal release or from a canary release.
+ * @return {!Promise<string>}
+ */
 function getLastGitTag() {
   var options = {
     args: 'describe --abbrev=0 --tags'
   };
-  return gitExec(options).then(function(tag) {
-    if (typeof tag == 'string') {
-      return tag.replace(/\n/, '');
-    }
-    throw new Error('No tag found.');
-  });
+  var canaryGrep = isCanary ? 'grep canary$' : 'grep -v canary$';
+  return exec('git tag | ' + canaryGrep + ' | ' +
+      'xargs -I@ git log --format=format:"%ai @%n" -1 @ | ' +
+      'sort -r | awk \'{print $4}\' | head -1').then(function(tag) {
+        return tag.replace('\n', '');
+      });
 }
 
+/**
+ * @param {string} tag
+ * @return {!Promise<string>}
+ */
 function getGitLog(tag) {
   var options = {
-    args: 'log release...' + tag + ' --pretty=format:%s --merges'
+    args: 'log ' + branch + '...' + tag + ' --pretty=format:%s --merges'
```

Comment:
@jridgewell thanks. will make the updates as i update this file (will be making some more this week)